### ISSUE TITLE
ofSystemLoadDialog and ofSystemSaveDialog error ?

### ISSUE THREAD

**Issue by eloimaduell** [openframeworks/openFrameworks#3348]
The error i guess is that draw is not drawing anymore on screen ... so maybe GLcontext is lost ? 

If i use OF 0.84 i got errors on both, load and save.
If i get the ofSystemUtils.cpp from Github and put it on the OF 0.84 the load dialog problem is solved, but not the load dialog problem ... 

I experienced this with XCode 5 and OSX 10.9.1 if that helps ...¿?

This is what i see if i resize the window when draw is not working ... nice GL glitch ;)
![screenshot 2014-10-23 02 37 54](https://cloud.githubusercontent.com/assets/178728/4746709/b0a89c5e-5a4d-11e4-98b7-95c93b4b64e3.jpg)

**Comment by timknapen** [comment]
I have the same issue with XCode6.3.2 OF 0.8.4 on OSX 10.10.3
The strange thing is that it only started happening after running the same app a few times! 
So the first few times I compiled and ran the app it worked fine, then all of a sudden it stopped working. I did see some weird shareKit errors popping up:
```
NSSoftLinking - The ShareKit framework's library couldn't be loaded from /System/Library/PrivateFrameworks/ShareKit.framework/Versions/A/ShareKit.
```
I'll try to investigate further.

**Comment by pizthewiz** [comment]
@timknapen if you have a spare moment and can get a solid repro case down, could you please try to reproduce on the master branch as quite a lot has changed since the 0.8.4.

**Comment by timknapen** [comment]
@pizthewiz yes, i'll try it as soon as possible

**Comment by timknapen** [comment]
@pizthewiz I dowloaded the 0.9.0 master branch as a zip, but it doesnt include an emptyExample for Xcode. Any idea if there's an example project I can start from to test this?

**Comment by arturoc** [comment]
it's easier to use the nightly builds, you can find them in http://openframeworks.cc/downloads

**Comment by timknapen** [comment]
thanks @arturoc I'll try that

**Comment by timknapen** [comment]
@arturoc is there no nightly build for OSX?

**Comment by arturoc** [comment]
yes: http://192.237.185.151/versions/nightly/of_v20150616_osx_nightly.zip

**Comment by timknapen** [comment]
@pizthewiz i tried with the main branch (the one @arturoc linked to). I can perfectly reproduce the problem in this version with this simple code:
``` c++
/// ofApp.h
#pragma once

#include "ofMain.h"

class ofApp : public ofBaseApp{
	
public:
	void setup();
	void update();
	void draw();
	void keyPressed(int key);
	bool bShowLoadDialog;
};
```

and

```c++
// ofApp.cpp
#include "ofApp.h"

//--------------------------------------------------------------
void ofApp::setup(){
	ofBackground(255);
	bShowLoadDialog = false;
}

//--------------------------------------------------------------
void ofApp::update(){
	
}

//--------------------------------------------------------------
void ofApp::draw(){
	ofSetColor(0);
	ofDrawBitmapString("Drawing frame nr "+ofToString(ofGetFrameNum()), 15, 15);

}

//--------------------------------------------------------------
void ofApp::keyPressed(int key){
	bShowLoadDialog = true;
	if(bShowLoadDialog){
		bShowLoadDialog = false;
		cout << "Show the file dialog" << endl;
		ofFileDialogResult dialogResult = ofSystemLoadDialog();
	}
}
``` 

after opening the load file dialog and closing it again (click cancel), the drawing of the OF view doesn't restart.

**Comment by timknapen** [comment]
The weird thing with ``` bShowLoadDialog = true;``` in keyPressed is because I first tried
```c++
if(bShowLoadDialog){
        bShowLoadDialog = false;
        cout << "Show the file dialog" << endl;
        ofFileDialogResult dialogResult = ofSystemLoadDialog();
    }
```
inside update();
I thought the p
